### PR TITLE
Jetpack Backup: Add selector to get backups_stopped flag

### PR DESCRIPTION
**Pull request by rustamshaikh**
PT: peaFOp-1b9-p2

## Proposed Changes

As a part of As a part of the [Project - Backup retention option on cancelation flow](1204174192068668-as-1204101785545874), we need to show this 2-day retention option to customers only who have reached their storage limit. To determine if the customer have reached the storage limit, we have added a `backups_stopped` flag to the WPCOM API and it will be used in the PR https://github.com/Automattic/wp-calypso/pull/75400 to show/hide the option

## Testing Instructions

Unit tests only:

```
➜  $ yarn test-client client/state/rewind/test/selectors.js                 
 PASS  client/state/rewind/test/selectors.js

Test Suites: 1 passed, 1 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        0.894 s, estimated 1 s
Ran all test suites matching /client\/state\/rewind\/test\/selectors.js/i.
```

```
➜  $ yarn test-client client/state/rewind/size/test        
 PASS  client/state/rewind/size/test/reducer.js

Test Suites: 1 passed, 1 total
Tests:       14 passed, 14 total
Snapshots:   0 total
Time:        0.578 s, estimated 1 s
Ran all test suites matching /client\/state\/rewind\/size\/test/i.
```

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/rewind/size/test/reducer.js [hunk 1]
```diff
@@ -82,4 +82,30 @@ describe( 'rewind.size reducers', () => {
 			).toEqual( 7 );
 		} );
 	} );
+	describe( 'backupsStopped', () => {
```
Comment (Initsogar) [inline client/state/rewind/size/test/reducer.js:62]: Sorry for nit, could we add an additional line break before the `describe` here?
```suggestion

	describe( 'backupsStopped', () => {
```

File: client/state/rewind/size/test/reducer.js [hunk 2]
```diff
@@ -82,4 +82,30 @@ describe( 'rewind.size reducers', () => {
...
+		it( 'should return false when there is nothing in the state', () => {
```
Comment (Initsogar) [inline client/state/rewind/size/test/reducer.js]: This scenario seems to test how the reducer will behave when the state is undefined, action type `REWIND_SIZE_SET` and also an empty size object. The current description seems a bit ambiguous as it only specifies the current state.

Maybe we could rephrase it to: `should return false when state is undefined and REWIND_SIZE_SET action is dispatched with no size object`.

What do you think?

File: client/state/rewind/size/test/reducer.js [hunk 3]
```diff
@@ -82,4 +82,30 @@ describe( 'rewind.size reducers', () => {
...
+			expect(
+				sizeReducer( undefined, { type: REWIND_SIZE_SET, size: {} } ).backupsStopped
```
Comment (Initsogar) [inline client/state/rewind/size/test/reducer.js]: Is it valid to dispatch `REWIND_SIZE_SET` with an empty `size` object?
I understand this action is dispatched when the `/rewind/size` endpoint returns 200.

If there's a chance that the endpoint might return 200 and don't include the `backupsStopped` flag, I think it's still okay to test it.

File: client/state/rewind/size/test/reducer.js [hunk 4]
```diff
@@ -82,4 +82,30 @@ describe( 'rewind.size reducers', () => {
...
+		} );
+		it( 'should return new the value when it is updated and the action is REWIND_SIZE_SET', () => {
+			expect(
+				sizeReducer(
+					{ backupsStopped: true },
+					{ type: REWIND_SIZE_SET, size: { backupsStopped: false } }
+				).backupsStopped
+			).toEqual( false );
+		} );
+		it( 'should return null when the action is not REWIND_SIZE_SET', () => {
```
Comment (Initsogar) [inline client/state/rewind/size/test/reducer.js]: I'd suggest to rephrase this description to `should return null when the action is not REWIND_SIZE_SET and the state is undefined`.

In addition, based on this test case, should we also test that it returns the `backupsStopped` value from state when `REWIND_SIZE_GET` action is dispatched?

**Review by Initsogar — COMMENTED** [review]
Looks great to me! Just left a few comments related to the reducer's unit tests that would be great if you could take a look.

File: client/state/rewind/size/test/reducer.js (hunk 3 above)
Comment (rustamshaikh) [reply client/state/rewind/size/test/reducer.js]: Right, I'll remove it 👍

File: client/state/rewind/size/test/reducer.js (hunk 4 above)
Comment (rustamshaikh) [reply client/state/rewind/size/test/reducer.js]: > In addition, based on this test case, should we also test that it returns the backupsStopped value from state when REWIND_SIZE_GET action is dispatched?

The `REWIND_SIZE_GET` action doesn't exist, wondering why we have added in other tests, I will remove them.

**Review by Initsogar — APPROVED** [review]
Looks good to me 👍🏽
Thanks for addressing the comments!

I restarted the E2E tests builds, and it seems they are working fine now.
